utils/window/window.go

```diff
@@ -95,18 +97,31 @@ func (w *window[T]) Length() int {
 	return w.elements.Len()
 }
 
+// EvictEvery evicts stale nodes from the window at a regular interval.
+func (w *window[T]) EvictEvery(ctx context.Context, freq time.Duration) {
+	ticker := time.NewTicker(freq)
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			w.lock.Lock()
+			w.removeStaleNodes()
+			w.lock.Unlock()
+		case <-ctx.Done():
+			return
+		}
+	}
+}
```

Comment:
Thoughts on just using a `Mutex` rather than a `RWMutex` and then calling `removeStaleNodes` at the start of every function invocation - rather than adding this periodic polling?

Reply:
Thanks -- that's way better.